### ISSUE TITLE
False positive "import six" should be placed before "import ssl" on Windows

### ISSUE THREAD

**Issue by andy-maier** [PyCQA/pylint#5943]
### Bug description

On Windows (and only there, not on Ubuntu or MacOS), we get:

C0411: third party import "import six" should be placed before "import ssl"

Since the "ssl" module in Python is also supported for Windows, this is a false positive.

Source file `ssl_import_order.py` to demonstrate the issue:

```python
"""
Test file.
"""

# pylint: disable=consider-using-f-string

from __future__ import print_function, absolute_import

import sys
import os
import ssl
import six


def main():
    """
    Some stupid code that uses the imports.
    """
    my_file = sys.argv[0]
    my_dir = os.path.dirname(my_file)
    print("My directory: {}".format(my_dir))

    assert isinstance(my_dir, six.string_types)

    print("OpenSSL version: {}".format(ssl.OPENSSL_VERSION))


if __name__ == '__main__':
    main()
```

### Command used

```shell
pylint -v ssl_import_order.py
```

### Pylint output

```shell
************* Module ssl_import_order
ssl_import_order.py:18:0: C0411: third party import "import six" should be placed before "import ssl" (wrong-import-order)
```

### Expected behavior

The import order should be accepted on Windows in the same way it is accepted on non-Windows.

### Pylint version

Here it gets interesting. I tried to increase the pylint version on Windows from our normal version in order to verify whether the issue is fixed. The interesting part is that we have two test runs, one with latest package versions that succeeds, and one with minimum package versions that fails with the issue reported here. In both cases, the pylint version is 2.12.2 and the astroid version is 2.9.3. So the difference in behavior must be caused by some other package.

* Run with latest package versions on Python 3.10 that succeeds:
  test run: https://github.com/pywbem/pywbem/runs/5623056094?check_suite_focus=true#step:18:15
  ```shell
  pylint --version
  pylint 2.12.2
  astroid 2.9.3
  Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
  ```

* Run with minimum package versions on Python 3.10 that fails:
  test run: https://github.com/pywbem/pywbem/runs/5623056261?check_suite_focus=true#step:18:16
  ```shell
  pylint --version
  pylint 2.12.2
  astroid 2.9.3
  Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
  ```

The test runs display the installed package versions.

### OS / Environment

Windows-latest on Github Actions, i.e. WIndows Server 2022

### Additional dependencies

See the display of packages in the test runs linked above.

**Comment by Pierre-Sassoulas** [comment]
Under (pylint's) hood there is [isort](https://github.com/PyCQA/isort). The fast way to fix this would be to modify the isort configuration, otherwise it's an isort issue first.

**Comment by andy-maier** [comment]
Thanks for the hint. `isort` is at 4.2.5 in the failing run, and at 5.10.1 in the succeeding run, so this provides some room for experiments!

**Comment by Pierre-Sassoulas** [comment]
I guess the solution here is to upgrade isort to have the latest features and bug fixes. We're trying to be permissive in the version of isort we support. People using isort 4.2.5 probably have good reasons to do so (?), and pylint is compatible with this version (but this version of isort is what it is).

We could maybe change pylint's supported version to 4.3.21 (latest 4. compatible version) ? isort 4 starts to be old this release is from June 2019. Users would have the latest isort still compatible with their un-migrated configuration.  What do you think ?

**Comment by andy-maier** [comment]
Our pywbem project is a library, so we do not know what the constraints of our users are. For that reason, we want to give them the largest possible range of versions for our dependent packages. We have a defined set of minimum versions of dependent packages, and we also test against these minimum versions. We increase these minimum versions driven by triggers such as bugs we encounter in the testing, or some other package or Python causing a need to increase it. That's why we would not increase the minimum version of isort to the latest 4.x just to pick up fixes for issues we have not encountered. Our users can still do that.

I found out that isort 4.2.8 solves the original issue, but it fails on Python 3.10 because it uses `collections.MutableSet` which no longer works there. isort 4.3.8 fixes that issue and works with Python 3.10 for pylint (in our project, pylint is the only user of isort). However, the official support for Python 3.10 comes only with isort 5.10.0 according to their change log.

So I think there are two options for our project:

* Option 1:

```
isort>=4.3.8
```

* Option 2:

```
isort>=4.2.8; python_version <= '3.9'
isort>=5.10.0; python_version >= '3.10'
```

I have tested both options with pywbem on Python 2.7, 3.6...3.10 and they both work (on py 3.5 we have other package install issues that hide any pylint issue at the moment). For the pylint/astroid versions that were used with these Python versions, see https://github.com/pywbem/pywbem/blob/andy/missing-constraints/dev-requirements.txt#L75.
Given that both work, we will probably use option 1.

For pylint, you can of course make a different decision for the versions of the isort dependency, I just wanted to explain the motivation for our project. If you use isort 4.3.21 as the minimum version for the next pylint version, that is fine with us, and we will then increase our new isort minimum version 4.3.8 accordingly once we pick up that pylint version.

**Comment by Pierre-Sassoulas** [comment]
I'm going to close as won't fix because I think it's better if users can choose what version of isort they use. You have to be explicit about it to use an old version of isort so if this bug happens it's voluntarily.
